### ISSUE TITLE
possible error relating to gdal

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1657]
On the "develop" branch, I get as shown below.  It may be just a matter of my own system, which could be messed up because of my tests with the upcoming `gdal` library, etc.

(I *think* this code will not be using any `sf` tests, but I could be wrong.)

@richardsc do you get this error with the latest on "develop"?

```
   > ### Aliases: subset,coastline-method
   > 
   > ### ** Examples
   > 
   > ## No test: 
   > library(oce)
   > data(coastlineWorld)
   > ## Eastern Canada
   > if (requireNamespace("raster", quietly=TRUE) && requireNamespace("rgeos", quietly=TRUE)) {
   +     cl <- subset(coastlineWorld, -80<lon & lon<-50 & 30<lat & lat<60)
   +     ## The plot demonstrates that the trimming is as requested.
   +     plot(cl, clon=-65, clat=45, span=6000)
   +     rect(-80, 30, -50, 60, bg="transparent", border="red")
   + }
   Assertion failed: (0), function query, file ../../../../src/geos-3.7.2/src/index/strtree/AbstractSTRtree.cpp, line 293.
✓  checking for unstated dependencies in ‘tests’ (1.1s)
```

**Comment by dankelley** [comment]
Weirdly,  I do not get this if I run the code manually, as below.  Then again, I am never really too sure what it means when RStudio's "check" operation reports a problem.
```R
library(oce)
data(coastlineWorld)
## Eastern Canada
if (requireNamespace("raster") && requireNamespace("sp")) {
    cl <- subset(coastlineWorld, -80<lon & lon<-50 & 30<lat & lat<60)
    ## The plot demonstrates that the trimming is as requested.
    plot(cl, clon=-65, clat=45, span=6000)
    rect(-80, 30, -50, 60, bg="transparent", border="red")
}
```

**Comment by richardsc** [comment]
I'll take a look. I just updated my Mac to 3.6.2 (it was at 3.5) so might take a little bit to get things installed.

To be clear, you mean for me to check using CRAN installs of all the oce dependencies, right? I don't need to try and install all the gdal system libraries, etc?

**Comment by dankelley** [comment]
Thanks!

Yes, please avoid messing with system libraries. I think my machine is messed up with them, because of trying to get ahead of `rgdal` changes.

Meantime, I'm going to see if I can make the problem go away by trying to do this coastline work with `sf`, which is an update for the `sp` that I'm using now.  (It could be in `raster` though, but, again, all I'm doing is a polygon intersection, and `sf` can do that, no problem.)

**Comment by richardsc** [comment]
Uh oh ... I shouldn't have upgraded ...

```
clang: warning: no such sysroot directory: '/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk'
In file included from RcppExports.cpp:4:
In file included from /Library/Frameworks/R.framework/Versions/3.6/Resources/library/Rcpp/include/Rcpp.h:27:
In file included from /Library/Frameworks/R.framework/Versions/3.6/Resources/library/Rcpp/include/RcppCommon.h:29:
In file included from /Library/Frameworks/R.framework/Versions/3.6/Resources/library/Rcpp/include/Rcpp/r/headers.h:67:
In file included from /Library/Frameworks/R.framework/Versions/3.6/Resources/library/Rcpp/include/Rcpp/platform/compiler.h:100:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/cmath:301:10: fatal error: 'math.h' file not found
#include <math.h>
         ^
1 error generated.
make: *** [RcppExports.o] Error 1
ERROR: compilation failed for package ‘oce’
```

Going to need to dig into this some more. This machine is 11 years old ...

**Comment by dankelley** [comment]
One more thing, to get the error you need to set an env var.  In RStudio that's done with a setting for "R check" in `oce.Rproj`.  I have this set to blank in "develop" for a while, because I want to test a fix for issue #1656 
.
<img width="547" alt="Screen Shot 2020-02-09 at 2 36 54 PM" src="https://user-images.githubusercontent.com/99469/74107790-0ca4a000-4b4a-11ea-9e9f-1e15f2086396.png">

**Comment by dankelley** [comment]
In "develop" commit 56c5453545f8d7fbf99bb713a76636594a75dd05 I changed so that this polygon-intersection calculation is done with the `sf` package, instead of the old method of using the `raster` and `sp` packages.  It builds and checks fine on my home machine.  I'm putting below a checklist.  When these are all done, I plan to just close the issue, because it will have become moot.  Since the tests take about 1/2 hour each, my guess is that I'll close this issue within an hour.

* [ ] local tests on DK's home machine, using RStudio build and check
* [ ] local tests on DK's home machine, using `R CMD check`
* [ ] nonlocal tests on travisCI
* [ ] nonlocal tests on appveyor

**Comment by dankelley** [comment]
All my tests indicate that "develop" commit 82437891e96f93bb31ae0feea3ebfbfd89001fe8 works fine on my home machine.  I will try it tomorrow on my work machine, and if it builds and checks there, I'll close this issue.

I like the side effect of this bug -- one more code dependence on `sp` and `raster` eliminated. It was just laziness that I was using calls to those two things to find the intersection of two polygons.

**Comment by richardsc** [comment]
Hm, ok ... I fixed my build issue (had to run an `xcode-select --install` from a terminal to update commandline tools), and then built, checked, and installed based on commit 70c3d3da2fd56f2 (prior to your update using `sf` from above).

I ran:
```bash
R CMD build oce
R CMD check --run-donttest oce_1.1-2.tar.gz
R CMD INSTALL oce_1.1-2.tar.gz
```

It all worked just fine.

Note -- I didn't use RStudio, because, honestly, I have found that it has caused more problems than it solves on my various machines (also because I basically just don't use it for any code/package development at all so it's a pain to fire it up just to build a package).

1. On this machine (ancient Mac) it does not seem to run reliably, and I've had issues with features in the release version in the past that required me to install (partially broken in other ways) devel versions.

2. On my linux virtual machine (Mint/Ubuntu) something about the upgrade to R3.5 broke the linux RStudio ability to build packages. 

3. On my WSL (windows subsystem for linux) RStudio does not run at all.

So, basically, I am just an old-fashioned `R CMD ...` package builder now.

**Comment by dankelley** [comment]
Thanks for the check.  I also include `R CMD` in my toolchain, but I start with RStudio mainly because I have it set up to build the `.Rd` files.  But I never push to GH without doing that, so you should never need to use RStudio for such updates (unless you change the Roxygen, but even then, no worries, because when I pull and rebuild, the Rd wil get updated.)

**Comment by dankelley** [comment]
Oh, I see I didn't propose an algorithm for closure.  Since it's been addressed -- I no longer get the error with the new code -- I'm closing this.

PS. I only ever got the error whilst doing a 'check' in RStudio, with the `--run-donttest` flag.  It did not occur when I actually ran the code. The error was a problem deep down in the C code for gdal (that's what an `assert` error is: a low-level C thing), and my only guess is that RStudio might have been using a different version of `geos`.  (I fear that my system is a bit messed up, because I was trying to get the not-yet-released rgdal working, using a non-standard gdal, etc.)
